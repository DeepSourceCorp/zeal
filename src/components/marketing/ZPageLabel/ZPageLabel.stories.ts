import ZPageLabel from './ZPageLabel.vue'

export default {
  title: 'Zeal Marketing/Page Label',
  component: ZPageLabel,
  excludeStories: /.*Data$/
}

export const DefaultWithoutText = () => ({
  components: { ZPageLabel },
  template: `<div class='padded-container'>
      <z-page-label></z-page-label>
    </div>`
})

export const PageLabelWithText = () => ({
  components: { ZPageLabel },
  template: `<div class='padded-container'>
      <z-page-label class="bg-green">Page</z-page-label>
    </div>`
})
