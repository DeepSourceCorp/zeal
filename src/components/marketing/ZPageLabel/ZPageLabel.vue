<template>
  <span 
    :class="[`${color && `z-gradient--${color}`}`]"
    class="inline-block px-2 text-sm text-vanilla-100 uppercase rounded-sm"
  >
    <slot>Page Label</slot>
  </span>
</template>

<script lang="ts">
export default {
  name: 'ZPageLabel',
  props: {
    color: {
      type: String,
      default: 'alpha'
    }
  }
}
</script>

<style lang="css" scoped>
/* Gradient - Alpha */
.z-gradient--alpha {
  @apply bg-gradient-to-br from-alpha-begin to-alpha-end;
}
</style>
