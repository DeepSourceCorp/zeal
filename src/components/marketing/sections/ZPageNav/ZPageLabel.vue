<template>
  <div class="inline-block px-2 text-sm uppercase rounded-sm">
    <slot>Page Label</slot>
  </div>
</template>

<script lang="ts">
export default {
  name: 'ZPageLabel',
  props: {
    title: String,
    articles: {
      type: Array,
      default: [
        {
          title: '',
          link: ''
        },
        {
          title: '',
          link: ''
        }
      ]
    }
  }
}
</script>
