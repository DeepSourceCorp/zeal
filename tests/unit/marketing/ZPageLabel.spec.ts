import Vue from 'vue'
import { createLocalVue, mount, Wrapper } from '@vue/test-utils'

import ZPageLabel from '../../../src/components/marketing/ZPageLabel'

describe('Page Label component', () => {
  let mountFunction: (options?: object) => Wrapper<Vue>
  let localVue: typeof Vue
  beforeEach(() => {
    localVue = createLocalVue()
    mountFunction = (options = {}) => {
      return mount(ZPageLabel, {
        localVue,
        ...options
      })
    }
  })
  it('renders a page label when the component is mounted', () => {
    expect(mountFunction().html()).toMatchSnapshot()
  })

  it('renders a page label with the given text', () => {
    let mountedInstance = mountFunction({
      slots: {
        default: 'Guide'
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
    expect(mountedInstance.text()).toContain('Guide')
  })

  it('renders a page label with default gradient background', () => {
    let mountedInstance = mountFunction({
      slots: {
        default: 'Guide'
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
    expect(mountedInstance.classes()).toContain('bg-gradient-dawn')
  })
  it('renders a page label with custom background color', () => {
    let mountedInstance = mountFunction({
      propsData: {
        backgroundColor: 'ink-400'
      },
      slots: {
        default: 'Guide'
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
    expect(mountedInstance.classes()).toContain('bg-ink-400')
  })

  it('renders a page label with default text color', () => {
    let mountedInstance = mountFunction({
      slots: {
        default: 'Guide'
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
    expect(mountedInstance.classes()).toContain('text-vanilla-100')
  })
  it('renders a page label with custom text color', () => {
    let mountedInstance = mountFunction({
      propsData: {
        textColor: 'slate'
      },
      slots: {
        default: 'Guide'
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
    expect(mountedInstance.classes()).toContain('text-slate')
  })
})
