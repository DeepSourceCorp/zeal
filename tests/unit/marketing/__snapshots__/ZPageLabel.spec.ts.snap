// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Label component renders a page label when the component is mounted 1`] = `<span class="inline-block px-2 text-sm uppercase rounded-sm bg-gradient-dawn text-vanilla-100">Page Label</span>`;

exports[`Page Label component renders a page label with custom background color 1`] = `<span class="inline-block px-2 text-sm uppercase rounded-sm bg-ink-400 text-vanilla-100">Guide</span>`;

exports[`Page Label component renders a page label with custom text color 1`] = `<span class="inline-block px-2 text-sm uppercase rounded-sm bg-gradient-dawn text-slate">Guide</span>`;

exports[`Page Label component renders a page label with default gradient background 1`] = `<span class="inline-block px-2 text-sm uppercase rounded-sm bg-gradient-dawn text-vanilla-100">Guide</span>`;

exports[`Page Label component renders a page label with default text color 1`] = `<span class="inline-block px-2 text-sm uppercase rounded-sm bg-gradient-dawn text-vanilla-100">Guide</span>`;

exports[`Page Label component renders a page label with the given text 1`] = `<span class="inline-block px-2 text-sm uppercase rounded-sm bg-gradient-dawn text-vanilla-100">Guide</span>`;
